scripts/core/editor3/components/spellchecker/SpellcheckerDecorator.tsx

```diff
@@ -20,45 +20,51 @@ export function getSpellcheckWarningsByBlock(
         return Promise.resolve({});
     }
 
-    const rangesByBlock: Array<{blockKey: string, startOffset: number, endOffset: number}> = [];
+    return ng.getService('spellcheck').then((spellcheck) => {
```

Comment:
`getSpellcheckWarningsByBlock` too should only be called when spellchecker is already enabled.

It's checked [here](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/core/editor3/components/Editor3Component.tsx#L148) and the property is coming from [here](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/core/editor3/store/index.ts#L141). To disable the spellchecker you need to update `store.spellchecker.enabled`.

Reply:
If you look [here](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/core/editor3/store/index.ts#L98), `spellcheckerDisabledInConfig` depends on the configuration settings. That means we can not enable and disable spellchecker from the UI. We can only decide if we want to run spellchecker or not using `Run automatically` toggle button in the UI. So, why to update the `store.spellchecker.enabled`?

Reply:
Yeah, the code is a bit confusing. `spellcheckerDisabledInConfig` and `Run automatically` refer to spellchecker in general. It can be either for tansa or our internal one. `store.spellchecker.enabled` is only controlling our internal one, which doesn't have the concept of "run automatically" - it's either on or off.

Why do you need to disable the spellchecker in the first place?

Reply:
I don't need to disable the spellchecker. I thought setting `store.spellchecker.enabled` to `false` would mean disabling the spellchecker as you mentioned in the second comment.
So, the conclusion is if we turn on `Run automatically` then `store.spellchecker.enabled` should be set to true and vice versa, right?

Reply:
> I don't need to disable the spellchecker.

What is this doing then?
```
if (spellcheck.isAutoSpellchecker !== true) {
    return Promise.resolve({});
}
```

> So, the conclusion is if we turn on Run automatically then store.spellchecker.enabled should be set to true and vice versa, right?

yes

Reply:
> What is this doing then?
 If `run automatically` is turned off `spellcheck.isAutoSpellchecker` is set to false and so, we don't need to call the `check` method for finding the errors.

Reply:
You should set `store.spellchecker.enabled` to false for this use case. Let me know if you have any more doubts about anything.